orc/src/main/java/org/apache/iceberg/orc/ExpressionToSearchArgument.java

```diff
@@ -270,7 +271,8 @@ public <T> Action notStartsWith(Bound<T> expr, Literal<T> lit) {
 
   @Override
   public <T> Action predicate(BoundPredicate<T> pred) {
-    if (UNSUPPORTED_TYPES.contains(pred.ref().type().typeId())) {
+    if (UNSUPPORTED_TYPES.contains(pred.ref().type().typeId())
+        || !(pred.term() instanceof BoundReference)) {
```

Comment:
Should evaluate on reference term only.

Reply:
The reason is that we support push-down transform functions now. However, the file format can't handle such filters. The reason for parquet too. Another choice is we don't push down those transform functions for unpartitioned tables.